cmd/find-main.go

```diff
@@ -0,0 +1,141 @@
+/*
+ * Minio Client (C) 2014, 2015 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/console"
+)
+
+// find specific flags
+var (
+	findFlags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "name",
+			Usage: "Find files of a matching name",
+		},
+		cli.StringFlag{
+			Name:  "path",
+			Usage: "Find global matches on the entire path",
+		},
+		cli.StringFlag{
+			Name:  "regex",
+			Usage: "Matches all files with PCRE regular expression",
+		},
+		cli.StringFlag{
+			Name:  "print",
+			Usage: "Prints the found files given args to STDOUT",
+		},
+		cli.StringFlag{
+			Name:  "mcexec",
```

Comment:
I think there is a lot of value in enhancing the semantics of substitution... So I will elaborate on the details so that we can more easily discuss the best answer. 

Today the following is allowed

{}                   Full pathname (with alias substituted for URL in object store)
{base}           Pathname after the last slash (essentially the file name or object name)
{dir}               Pathname up to the last slash (full pathname without the file or object name)

If 'mc find' is executed with a target of an object store then the pathnames returned (today) will have the "alias" substituted for the "URL"... for example:

on my system, myminio is at 'http://169.254.194.86:9000'

>mc find myminio -name *.jpg 
myminio/copy-images/Blinds.jpg
myminio/copy-images/Bokeh_Tails.jpg
...

The find output is easier to read and more intuitive it it returns names with the alias. Those names are not valid outside of mc. 

It makes sense that {} substitute the output from the find command (the full path). It might make sense to have a separate flag to control the output format (full URL or with alias). 

Independently, I kind of like the idea of have --mcexec (which always takes an mc command) and --exec (which can take any command).  The user will just have a syntax available to get full pathnames (as full URLS) and will have to be aware that only some shell commands will be able to use those URLs. (your example of wget makes sense, but lots of other commands will not work even with the "full" URL).